### PR TITLE
969 improve errors 1

### LINKED ISSUE #984: DB RW check is faulty, it shouldn't have an ! on the front:
The DB RW check is faulty, it shouldn't have an ! on the front:  

``` 
if !file_permissions_readonly(&i_meta) {
                warn!("WARNING: DB folder permissions on {} indicate it may not be RW. This could cause the server start up to fail!", db_par_path_buf.to_str()
                .unwrap_or_else(|| "<db_par_path_buf invalid>")
                );
            }
```

_Originally posted by @yaleman in https://github.com/kanidm/kanidm/issues/972#issuecomment-1207504519_

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #984, Improvements for #969. This properly swaps to tracing forest (hooray!), changes a few annoying errors, fixes the ro mode, and breaks out some of the tracing elements to a seperate crate to help build times and general layout.

- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: kanidmd/score/src/lib.rs [hunk 1]
```diff
@@ -548,6 +548,12 @@ pub fn recover_account_core(config: &Configuration, name: &str) {
 pub async fn create_server_core(config: Configuration, config_test: bool) -> Result<(), ()> {
     // Until this point, we probably want to write to the log macro fns.
 
+    error!(
```
Comment (yaleman) [inline kanidmd/score/src/lib.rs:551]: I'm guessing this is just a leftover from testing?

File: kanidm_unix_int/src/daemon.rs [hunk 2]
```diff
@@ -380,302 +379,313 @@ async fn main() {
...
-        let db_path = PathBuf::from(cfg.db_path.as_str());
-        // We only need to check the parent folder path permissions as the db itself may not exist yet.
-        if let Some(db_parent_path) = db_path.parent() {
-            if !db_parent_path.exists() {
+    tracing_forest::worker_task()
+        .set_global(true)
+        // Fall back to stderr
+        .map_sender(|sender| sender.or_stderr())
+        .build_on(|subscriber| subscriber
+            .with(EnvFilter::try_from_default_env()
```
Comment (yaleman) [inline kanidm_unix_int/src/daemon.rs:387]: does the logging config in `server.toml` do anything? it seems to have been been broken for a while...

**Review by yaleman — APPROVED** [review]
woooooo logging. 💜

File: kanidm_unix_int/src/daemon.rs (hunk 2 above)
Comment (Firstyear) [reply kanidm_unix_int/src/daemon.rs:387]: Part of the goal with the tagging stuff is to add that functionality back.

Comment (yaleman) [reply kanidm_unix_int/src/daemon.rs:387]: Sweeeeet, just confirming :D

**Pull request merged**